app/_hub/kong-inc/openid-connect/index.md

```diff
@@ -1071,6 +1242,203 @@ Default           | Required
+token endpoint request headers.
+
+
+#### config.token_headers_client
+
+When Kong calls token endpoint, you can specify the headers
+that Kong will pass to token endpoint from client request
+headers.
+
+
+#### config.token_headers_replay
+
+When Kong calls token endpoint, you can specify the headers
+(from the token endpoint response) that you want the Kong
+to send back to the client.
+
+
+#### config.token_headers_prefix
+
+If you want you can prefix the token endpoint headers with
+a string to differentiate them from other headers.
+
+
+#### config.token_headers_grants
+
+You can limit the grants for which the token headers are replayed:
+
+* `"password"`
+* `"client_credentials"`
+* `"authorization_code"`
+
+
+#### config.token_post_args_names
+
+Extra arguments names that you want to include in token
+endpoint post args.
+
+
+#### config.token_post_args_values
+
+Values for the extra arguments that you want to include
+in token endpoint post args.
+
+
+#### config.token_post_args_client
+
+Extra dynamic arguments that are passed from client to token
+endpoint.
+
+
+#### config.introspection_endpoint
+
+OAuth 2.0 introspection endpoint is not defined in OpenID Connect
+standards or OpenID Connect Discovery. Still some OpenID providers
+announce the introspection endpoint in discovery document. Kong
+tries hard to figure out the introspection endpoint, but sometimes
+it is not possible through automation. With this configuration
+parameter you can specify a URL for your issuer's introspection
+endpoint.
+
+
+#### config.introspection_hint
+
+Some providers such as `WSO2` don't implement OAuth 2.0 introspection
+standard correctly, and they want to have `token_type_hint` argument
+with value `bearer` instead of `access_token` when introspecting
+access tokens. With this parameter you can change the value of the
+`token_type_hint` argument that the plugin will pass to introspection
+endpoint when introspecting access tokens.
+
+
+#### config.introspection_headers_names
+
+Extra arguments names that you want to include in introspection
+request headers (e.g. use this to override `Authorization` header).
+
+
+#### config.introspection_headers_values
+
+Values for the extra arguments that you want to include in
+introspection request headers.
+
+
+#### config.introspection_post_args_names
+
+Extra arguments names that you want to include in introspection
+endpoint post args.
+
+
+#### config.introspection_post_args_values
+
+Values for the extra arguments that you want to include
+in introspection endpoint post args.
+
+
+#### config.introspection_post_args_client
+
+Extra dynamic arguments that are passed from client to introspection
+endpoint.
+
+
+#### config.introspect_jwt_tokens
+
+You can turn this option on (`true`) to enable introspection of
+JWT Bearer tokens. This enables checking of revoked JWT tokens,
+but it adds latency to request. So if you enable this, it is good
+to make Kong to IdP traffic as fast as possible. We still cache
```

Comment:
@bungle I'm confused here--It is a good idea to make Kong _listen_ to IdP traffic? I think the verb is missing here.

Reply:
I removed the whole sentence. It is already said in previous that it will add latency. It is not about Kong listening IdP traffic, what I meant there was just that user should ensure that kong to idp latency is as small as possible when Kong makes http calls to IdP endpoint because all that latency will also be added to proxied request (though as explained later, we do cache if configured so, and thus it affects the most on the cold node).